src/App.js

```diff
@@ -1,19 +1,42 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
 
 const App = () => {
+  const [chatData, setChatData] = useState(chatMessages);
+
+  const updateChatData = (updatedChatData) => {
+    setChatData(
+      chatData.map((entry) =>
+        entry.id === updatedChatData.id ? updatedChatData : entry
```

Comment:
If we passed up just the `id` from ChatEntry, the map would be identical (we still need to copy the array), but instead of simply picking between one of two objects, if we found the matching id, we would return the spread and modified version of entry (as we did with message in the current implementation of ChatEntry).

```js
        entry.id === id ? { ...entry, liked: !entry.liked } : entry
```